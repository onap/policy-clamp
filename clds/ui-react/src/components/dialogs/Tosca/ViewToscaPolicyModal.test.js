/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2020 AT&T Intellectual Property. All rights
 *                             reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import ViewToscaPolicyModal from './ViewToscaPolicyModal';
import { mount } from 'enzyme';


describe('Verify ViewToscaPolicyModal', () => {
	beforeEach(() => {
		fetch.resetMocks();
	});

	it('Test API Successful', () => {
		fetch.mockImplementationOnce(() => {
			return Promise.resolve({
				ok: true,
				status: 200,
				json: () => {
					return Promise.resolve({
						"index": "1",
						"policyModelTosca":"TCA",
						"policyModelType":"onap.policies.monitoring.cdap.tca.hi.lo.app",
						"version":"1.0.0",
						"policyAcronym": "TCA",
						"updatedDate": "2020-01-31T20:49:48.658795600Z",
						"updatedBy": "admin"
					});
				}
			});
		});
		const component = shallow(<ViewToscaPolicyModal/>);
	});

  it('Test API Exception', () => {
		fetch.mockImplementationOnce(() => {
			return Promise.resolve({
				ok: false,
				status: 500,
				json: () => {
					return Promise.resolve({
						"index": "1",
						"policyModelTosca":"TCA",
						"policyModelType":"onap.policies.monitoring.cdap.tca.hi.lo.app",
						"version":"1.0.0",
						"policyAcronym": "TCA",
						"updatedDate": "2020-01-31T20:49:48.658795600Z",
						"updatedBy": "admin"
					});
				}
			});
		});
		const component = shallow(<ViewToscaPolicyModal/>);
	});

	it('Test API Rejection', () => {
		const myMockFunc  = fetch.mockImplementationOnce(() => Promise.reject('error'));
		setTimeout(
    () =>
      myMockFunc().catch(e => {
        console.info(e);
      }),
    100
  );
	const component = shallow(<ViewToscaPolicyModal/>);
	expect(myMockFunc.mock.calls.length).toBe(1);
	});


	it('Test the tosca model view render method', () => {
		fetch.mockImplementationOnce(() => {
			return Promise.resolve({
				ok: true,
				status: 200,
				json: () => {
					return Promise.resolve({
						"index": "1",
						"policyModelTosca":"TCA",
						"policyModelType":"onap.policies.monitoring.cdap.tca.hi.lo.app",
						"version":"1.0.0",
						"policyAcronym": "TCA",
						"updatedDate": "2020-01-31T20:49:48.658795600Z",
						"updatedBy": "admin"
					});
				}
			});
		});
		const component = shallow(<ViewToscaPolicyModal/>);
		component.setState({ toscaNames: {
			"index": "1",
			"policyModelTosca":"TCA",
			"policyModelType":"onap.policies.monitoring.cdap.tca.hi.lo.app",
			"version":"1.0.0",
			"policyAcronym": "TCA",
			"updatedDate": "2020-01-31T20:49:48.658795600Z",
			"updatedBy": "admin"
		}
	});
  expect(component).toMatchSnapshot();
  });

	it('Test Table icons', () => {
		fetch.mockImplementationOnce(() => {
			return Promise.resolve({
				ok: true,
				status: 200,
				json: () => {
					return Promise.resolve({
						"index": "1",
						"policyModelTosca":"TCA",
						"policyModelType":"onap.policies.monitoring.cdap.tca.hi.lo.app",
						"version":"1.0.0",
						"policyAcronym": "TCA",
						"updatedDate": "2020-01-31T20:49:48.658795600Z",
						"updatedBy": "admin"
					});
				}
			});
		});
		const component = mount(<ViewToscaPolicyModal/>);
		expect(component.find('[className="MuiSelect-icon MuiTablePagination-selectIcon"]')).toBeTruthy();
  });

	it('Test handleYamlContent', () => {
		fetch.mockImplementationOnce(() => {
			return Promise.resolve({
				ok: true,
				status: 200,
				json: () => {
					return Promise.resolve({
						"index": "1",
						"policyModelTosca":"TCA",
						"policyModelType":"onap.policies.monitoring.cdap.tca.hi.lo.app",
						"version":"1.0.0",
						"policyAcronym":"TCA",
						"updatedDate": "2020-01-31T20:49:48.658795600Z",
						"updatedBy": "admin"
					});
				}
			});
		});
		const yamlContent = 'TCA Tosca model details';
		const component = shallow(<ViewToscaPolicyModal/>);
		component.find('[value="Please select Tosca model to view the details"]').prop('onChange')({ target: { value: yamlContent }});
    expect(component.state('content')).toEqual(yamlContent);
  });

	it('Test handleClose', () => {
		fetch.mockImplementationOnce(() => {
			return Promise.resolve({
				ok: true,
				status: 200,
				json: () => {
					return Promise.resolve({
						"index": "1",
						"policyModelTosca":"TCA",
						"policyModelType":"onap.policies.monitoring.cdap.tca.hi.lo.app",
						"version":"1.0.0",
						"policyAcronym": "TCA",
						"updatedDate": "2020-01-31T20:49:48.658795600Z",
						"updatedBy": "admin"
					});
				}
			});
		});
		const historyMock = { push: jest.fn() };
		const handleClose = jest.spyOn(ViewToscaPolicyModal.prototype,'handleClose');
		const component = shallow(<ViewToscaPolicyModal history={historyMock} />)
		component.find('[variant="secondary"]').prop('onClick')();
		expect(handleClose).toHaveBeenCalledTimes(1);
		expect(component.state('show')).toEqual(false);
		expect(historyMock.push.mock.calls[0]).toEqual([ '/']);
		handleClose.mockClear();
	});
});
