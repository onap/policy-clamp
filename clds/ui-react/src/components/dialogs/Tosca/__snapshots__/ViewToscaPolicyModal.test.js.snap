// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify ViewToscaPolicyModal Test the tosca model view render method 1`] = `
<Styled(Bootstrap(Modal))
  backdrop="static"
  keyboard={false}
  onHide={[Function]}
  show={true}
  size="xl"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  />
  <ModalBody>
    <WithStyles(Component)
      columns={
        Array [
          Object {
            "cellStyle": Object {
              "border": "1px solid black",
            },
            "field": "index",
            "headerStyle": Object {
              "backgroundColor": "#ddd",
              "border": "2px solid black",
            },
            "render": [Function],
            "title": "#",
          },
          Object {
            "cellStyle": Object {
              "border": "1px solid black",
            },
            "field": "policyModelType",
            "headerStyle": Object {
              "backgroundColor": "#ddd",
              "border": "2px solid black",
            },
            "title": "Policy Model Type",
          },
          Object {
            "cellStyle": Object {
              "border": "1px solid black",
            },
            "field": "policyAcronym",
            "headerStyle": Object {
              "backgroundColor": "#ddd",
              "border": "2px solid black",
            },
            "title": "Policy Acronym",
          },
          Object {
            "cellStyle": Object {
              "border": "1px solid black",
            },
            "field": "version",
            "headerStyle": Object {
              "backgroundColor": "#ddd",
              "border": "2px solid black",
            },
            "title": "Version",
          },
          Object {
            "cellStyle": Object {
              "border": "1px solid black",
            },
            "field": "updatedBy",
            "headerStyle": Object {
              "backgroundColor": "#ddd",
              "border": "2px solid black",
            },
            "title": "Uploaded By",
          },
          Object {
            "cellStyle": Object {
              "border": "1px solid black",
            },
            "editable": "never",
            "field": "updatedDate",
            "headerStyle": Object {
              "backgroundColor": "#ddd",
              "border": "2px solid black",
            },
            "title": "Uploaded Date",
          },
        ]
      }
      data={Array []}
      icons={
        Object {
          "FirstPage": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "LastPage": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "NextPage": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "PreviousPage": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "ResetSearch": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "Search": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "SortArrow": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        }
      }
      onRowClick={[Function]}
      options={
        Object {
          "headerStyle": Object {
            "backgroundColor": "#ddd",
            "border": "1px solid black",
            "fontSize": "15pt",
            "text": "bold",
          },
          "rowStyle": [Function],
        }
      }
      title="View Tosca Policy Models"
    />
    <div>
      <styled.textarea
        onChange={[Function]}
        value="Please select Tosca model to view the details"
      />
    </div>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="secondary"
    >
      Close
    </Button>
  </ModalFooter>
</Styled(Bootstrap(Modal))>
`;
